update error-chain to 0.7

pfernie opened the pull request:
Changes the error_chain! invocation to the new style used in error-chain 0.7, and bumps the dependency version

tafia commented on Cargo.toml:
```diff
@@ -15,7 +15,7 @@ quick-xml = "0.4.2"
 encoding = "0.2.33"
 byteorder = "0.5.3"
-error-chain = "0.6.2"
+error-chain = "0.7"
```
If possible I'd prefer `0.7.2`.

tafia commented on src/errors.rs:
```diff
@@ -8,23 +8,23 @@
-    }
+//    types {
+//        Error, ErrorKind, ResultExt, Result;
+//    }
```
Could you remove commented lines (it is confusing as `ResultExt` won't mean anything).

pfernie commented:
I updated the PR per your requests; I removed all commented code in errors.rs as it's just the example invocation from error-chain docs, and they are out of date w.r.t. 0.7.2.

tafia commented:
Thanks! 
Publishing 0.3.3

tafia commented:
> Publishing 0.3.3

upgraded byteorder to 1.0.0 too
